pkg/registry/resource/resourceclaim/strategy_test.go

```diff
@@ -141,22 +188,34 @@ const (
 )
 
 func TestStrategy(t *testing.T) {
-	if !Strategy.NamespaceScoped() {
+	fakeClient := fake.NewSimpleClientset()
+	mockNSClient := fakeClient.CoreV1().Namespaces()
+	strategy := NewStrategy(legacyscheme.Scheme, names.SimpleNameGenerator, mockNSClient)
+	if !strategy.NamespaceScoped() {
 		t.Errorf("ResourceClaim must be namespace scoped")
 	}
-	if Strategy.AllowCreateOnUpdate() {
+	if strategy.AllowCreateOnUpdate() {
 		t.Errorf("ResourceClaim should not allow create on update")
 	}
+	if strategy.nsClient == nil {
+		t.Errorf("ResourceClaim nsClient should not be nil")
+	}
 }
 
 func TestStrategyCreate(t *testing.T) {
 	ctx := genericapirequest.NewDefaultContext()
-
+	fakeClient := fake.NewSimpleClientset()
+	mockNSClient := fakeClient.CoreV1().Namespaces()
+	err := CreateNamespace(mockNSClient)
+	if err != nil {
+		t.Fatalf("failed to create test namespaces: %v", err)
+	}
 	testcases := map[string]struct {
 		obj                   *resource.ResourceClaim
 		adminAccess           bool
 		expectValidationError bool
 		expectObj             *resource.ResourceClaim
+		nsClient              v1.NamespaceInterface
```

Comment:
I would expect assertions to be made against the actions taken by the fake client.

Reply:
updated to `fakeClient := fake.NewSimpleClientset(ns1, ns2)` therefore no action will be taken after the mockClient is created. this is no longer applicable.

Reply:
This is still applicable, make the assertion against `ns, err := nsClient.Get(ctx, namespaceName, metav1.GetOptions{})`

Reply:
addressed in https://github.com/kubernetes/kubernetes/pull/130225/commits/d78d2b7fe072ffaf9c0be29f549ea8b59a8429c1